### PR TITLE
MGMT-7927 - Add assisted-test-infra v1 job for ocm-2.3 and ocm-2.4

### PR DESCRIPTION
**Pull request by eliorerz**
/cc @osherdp

### CODE REVIEW

File: ci-operator/config/openshift/assisted-test-infra/openshift-assisted-test-infra-v1.yaml [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+  steps:
+    cluster_profile: packet-assisted
+    test:
+    - ref: baremetalds-packet-check
+- as: sno-packet-check
+  cron: 0 */2 * * *
+  steps:
+    cluster_profile: packet-sno
+    test:
+    - ref: baremetalds-packet-check
```
Comment (lranjbar) [inline ci-operator/config/openshift/assisted-test-infra/openshift-assisted-test-infra-v1.yaml:65]: We don't need to duplicate these jobs so can remove.

File: ci-operator/config/openshift/assisted-test-infra/openshift-assisted-test-infra-v1.yaml [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+    tag: assisted-service
+  dev-scripts:
+    name: test
+    namespace: ocp-kni
+    tag: dev-scripts
+build_root:
+  image_stream_tag:
+    name: release
+    namespace: openshift
+    tag: golang-1.13
```
Comment (lranjbar) [inline ci-operator/config/openshift/assisted-test-infra/openshift-assisted-test-infra-v1.yaml:26]: Can we use golang-1.16 here?

File: ci-operator/config/openshift/assisted-test-infra/openshift-assisted-test-infra-ocm-2.3.yaml [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+    name: ocm-2.3
+    namespace: openshift
+    tag: assisted-installer-agent
+  assisted-installer-controller:
+    name: ocm-2.3
+    namespace: openshift
+    tag: assisted-installer-controller
+  assisted-service:
+    name: ocm-2.3
+    namespace: open-cluster-management
```
Comment (lranjbar) [inline ci-operator/config/openshift/assisted-test-infra/openshift-assisted-test-infra-ocm-2.3.yaml:16]: `namespace: openshift`

File: ci-operator/config/openshift/assisted-test-infra/openshift-assisted-test-infra-release-ocm-2.4.yaml [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+- dockerfile_path: Dockerfile.test-infra
+  from: src
+  inputs:
+    assisted-service:
+      as:
+      - quay.io/ocpmetal/assisted-service:latest
+  to: assisted-test-infra
+promotion:
+  name: assisted-test-ocm-2.4
+  namespace: open-cluster-management
```
[the remark this answers is not in the document]
Comment (eliorerz) [reply ci-operator/config/openshift/assisted-test-infra/openshift-assisted-test-infra-release-ocm-2.4.yaml]: Done

File: ci-operator/jobs/openshift/assisted-test-infra/openshift-assisted-test-infra-release-ocm-2.4-periodics.yaml [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    skip_cloning: true
+  extra_refs:
+  - base_ref: release-ocm-2.4
+    org: openshift
+    repo: assisted-test-infra
+  labels:
+    ci.openshift.io/generator: prowgen
+    pj-rehearse.openshift.io/can-be-rehearsed: "true"
+  name: periodic-ci-openshift-assisted-test-infra-release-ocm-2.4-e2e-metal-assisted-periodic
+  spec:
```
Comment (osherdp) [inline ci-operator/jobs/openshift/assisted-test-infra/openshift-assisted-test-infra-release-ocm-2.4-periodics.yaml:16]:
```suggestion
  reporter_config:
    slack:
      channel: '#assisted-deployment-ci'
      job_states_to_report:
      - failure
      - error
      report_template: '{{ if eq .Status.State "success" }} :green_jenkins_circle:
        {{ else }} :red_jenkins_circle: {{ end }} Job *{{.Spec.Job}}* ended with *{{.Status.State}}*.
        <{{.Status.URL}}|View logs>'
  spec:
```

File: ci-operator/jobs/openshift/assisted-test-infra/openshift-assisted-test-infra-ocm-2.3-periodics.yaml [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+    skip_cloning: true
+  extra_refs:
+  - base_ref: ocm-2.3
+    org: openshift
+    repo: assisted-test-infra
+  labels:
+    ci.openshift.io/generator: prowgen
+    pj-rehearse.openshift.io/can-be-rehearsed: "true"
+  name: periodic-ci-openshift-assisted-test-infra-ocm-2.3-e2e-metal-assisted-periodic
+  spec:
```
Comment (osherdp) [inline ci-operator/jobs/openshift/assisted-test-infra/openshift-assisted-test-infra-ocm-2.3-periodics.yaml:16]:
```suggestion
  reporter_config:
    slack:
      channel: '#assisted-deployment-ci'
      job_states_to_report:
      - failure
      - error
      report_template: '{{ if eq .Status.State "success" }} :green_jenkins_circle:
        {{ else }} :red_jenkins_circle: {{ end }} Job *{{.Spec.Job}}* ended with *{{.Status.State}}*.
        <{{.Status.URL}}|View logs>'
  spec:
```

**Pull request merged**